rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -299,6 +302,96 @@ void gs_frame::flip(draw_context_t, bool /*skip_frame*/)
 	}
 }
 
+void gs_frame::take_screenshot(std::vector<u8> ss_data, const u32 ss_width, const u32 ss_height, bool vulkan_buf)
+{
+	std::thread(
+	    [ss_width, ss_height, vulkan_buf](std::vector<u8> sshot_data) {
+		    std::string screen_path = fmt::format("%s/Screenshots/", fs::get_config_dir());
+
+		    if (!fs::is_dir(screen_path))
+		    {
+			    if (!fs::create_dir(screen_path))
+			    {
+				    LOG_ERROR(GENERAL, "Failed to create screenshot path: %s", screen_path);
+				    return;
+			    }
+		    }
+
+		    u32 file_index;
+		    for (file_index = 0; file_index < 100000000; file_index++)
+		    {
+			    std::string filename = fmt::format("%sScreenshot-%08d.png", screen_path, file_index);
+			    if (fs::is_file(filename))
+				    continue;
+
+			    fs::file ss_file(filename, fs::open_mode::create + fs::open_mode::write);
+			    if (!ss_file)
```

Comment:
if you open the file with fs::open_mode::excl you wont need to check if it exists already.

Reply:
Yeah but I want to differentiate between the case where the file already exist and the creation of the file just failed.

Reply:
for that you can read fs::g_tls_error to get the accurate error after failure.

Reply:
anything else than fs::error::exist indicates failure to save, if ::exist just continue;